misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -75,13 +72,14 @@ class FakeJobQueue @Inject constructor(
     attributes: Map<String, String>
   ) {
     val id = tokenGenerator.generate("fakeJobQueue")
-    val job = FakeJob(queueName, id, idempotenceKey, body, attributes, deliveryDelay)
-    jobQueues.getOrPut(queueName, ::ConcurrentLinkedDeque).add(job)
+    val job =
+        FakeJob(queueName, id, idempotenceKey, body, attributes, clock.instant(), deliveryDelay)
+    jobQueues.getOrPut(queueName, ::PriorityBlockingQueue).add(job)
   }
 
   fun peekJobs(queueName: QueueName): List<Job> {
     val jobs = jobQueues[queueName]
-    return jobs?.toList() ?: listOf()
+    return jobs?.sortedBy { it.id } ?: listOf()
```

Comment:
When we peek, we typically want to peek in the order the job was enqueued rather than in order of execution.